src/App.tsx

```diff
@@ -1,24 +1,64 @@
 import React from 'react';
+import { Clock } from './Components/Clock';
+
 import './App.scss';
 
-const App: React.FC = () => {
-  const timerId: NodeJS.Timer = setInterval(() => {
-    const date: Date = new Date();
-
-    // eslint-disable-next-line
-    console.log(date.toLocaleTimeString());
-  }, 1000);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-      <p>
-        Current time:
-        {' '}
-        {/* Print the time here instead of DevTools */}
-      </p>
-    </div>
-  );
+type State = {
+  isClockVisible: boolean,
+  clockName: number,
 };
 
+class App extends React.Component<{}, State> {
+  state = {
+    isClockVisible: true,
+    clockName: 0,
+  };
+
+  render() {
+    const { isClockVisible, clockName } = this.state;
+
+    return (
+      <div className="App">
+        <h1>React clock</h1>
+
+        {isClockVisible && <Clock name={clockName} />}
+
+        <button
+          className="App__button"
+          type="button"
+          onClick={() => {
+            this.setState({ isClockVisible: true });
+          }}
+        >
+          Show
+        </button>
+
+        <button
+          className="App__button"
+          type="button"
+          onClick={() => {
+            this.setState({ isClockVisible: false });
+          }}
+        >
+          Hide
+        </button>
+
+        <button
+          className="App__button"
+          type="button"
+          onClick={() => {
+            if (isClockVisible) {
+              this.setState({
+                clockName: Math.round(Math.random() * 1000),
+              });
+            }
```

Comment:
make a function for it, declaring whole function body in onClick handler is not a good idea